syntax = "proto3";

//namespace
package manager;

message error_info
{
    sint32 errorid=1;
    string errormsg=2;
}

message rsp_action_info
{
    int32 requestid=1;
    error_info info=2;
}

message rsp_head_info
{
    int32 requestid=1;
    error_info info=2;
    bool islast=3;
}

message sys_auth
{
    int32 authid = 1;
    string name = 2;
}

message role_auth
{
	int32 roleid = 1;
	repeated sys_auth auth = 2;
}

message req_login
{
    int32 requestid=1;
    string userid = 2;
    string pwd = 3;
}

message rsp_login
{
    rsp_head_info header=1;
    repeated role_auth auth = 2;
}

message req_sys_subs
{
    int32 requestid=1;
    string userid = 2;
}

message sys_user
{
    int32 id = 1;
    string userid = 2;
    string username = 3;
    string father = 4;
    int32 roleid = 5;
    string pwd = 6;
    string desc = 7;
    repeated sys_auth auths = 8;
}

message rsp_sys_subs
{
    rsp_head_info header=1;
    repeated sys_user info = 2;
}

message req_add_sys_user
{
    int32 requestid=1;
    sys_user user = 2;
}

message req_del_sys_user
{
    int32 requestid=1;
    string userid = 2;
}

message req_edit_sys_user
{
    int32 requestid=1;
    sys_user user = 2;
}

message sys_investor
{
    string investor = 1;
    string name = 2;
    int32 state = 3;
}

message req_risk_investors
{
    int32 requestid=1;
    string userid = 2; //riskid
}

message rsp_risk_investors
{
    rsp_head_info header=1;
    string userid = 2; //riskid
    repeated sys_investor investors = 3;
}

message req_investors
{
    int32 requestid=1;
    int32 state = 2;
}

message rsp_investors
{
    rsp_head_info header=1;
    repeated sys_investor investors = 2;
}

message req_add_investors
{
    int32 requestid=1;
    string userid = 2; //riskid
    repeated string investor = 3;
}

message req_del_investors
{
    int32 requestid=1;
    string userid = 2; //riskid
    repeated string investor = 3;
}

message rtn_change_tree
{
    //operatorid comment
    //1:add user 2:del user 3:edit user
    int32 operatorid = 1; 
    sys_user info = 2;
}

message rtn_change_risker_investor
{
    //operatorid comment
    //1:insert risker investor 2:remove risker investor
    int32 operatorid = 1; 
    string userid = 2;
    repeated string investor = 3;
}

message rtn_change_investor
{
    //operatorid comment
    //1:insert investor 2:remove investor
    int32 operatorid = 1; 
    repeated string investor = 2;
}

message insmrgnrate_info
{
    string tradingday=1;
    string contractcode=2;
    string exchangecode=3;
    string direction=4;
    string hedgeflag=5;
    double marginrate=6;
    double segmrgn=7;
}

message req_insmrgnrate
{
    int32 requestid=1;
}

message rsp_insmrgnrate
{
    rsp_head_info header=1;
    repeated insmrgnrate_info infos = 2;
}

message req_edit_insmrgnrate
{
    int32 requestid=1;
    repeated insmrgnrate_info infos = 2;
}

message rtn_change_insmrgnrate
{
    repeated insmrgnrate_info infos = 1;
}

message req_add_insmrgnrate
{
    int32 requestid=1;
    insmrgnrate_info info = 2;
}

message rtn_add_insmrgnrate
{
    insmrgnrate_info info = 1;
}

message req_del_insmrgnrate
{
    int32 requestid=1;
    insmrgnrate_info info = 2;
}

message rtn_del_insmrgnrate
{
    insmrgnrate_info info = 1;
}

message req_option_argv
{
    int32 requestid=1;
}

message option_argv
{
    int32 price_type=1; //1:lastprice 2:presettlementprice 3:precloseprice 4:settlementprice
    double vol=2;
    double rate=3;
    int32 step=4;
    int32 iterations=5;
}

message rsp_option_argv
{
    rsp_head_info header=1;
    option_argv info=2;
}

message req_option_argv_adj
{
    int32 requestid=1;
    option_argv info=2;
}

message req_contract
{
    int32 requestid=1;
}

message contract_info
{
    string          contractcode=1;
    string          productcode=2;
    string          productkind=3;
    string          exchangecode=4;
    string          marketcode=5;
    string          createdate=6;
    string          opendate=7;
    string          expiredate=8;
    int32           multiplier=9;
    double          pricetick=10;
    string          contractname=11;
    string          delivbegindate=12;
    string          delivenddate=13;
    string          excontractname=14;
    string          exproductname=15;
    string          ismrgndisc=16;
    string          productclass=17;
    double          baseprice=18;
    string          optionstype=19;
    double          strikeprice=20;
    string          underlyinginstrid=21;
    string          strikemode=22;
    double          underlyingmultiple=23;
}

message rsp_contract
{
    rsp_head_info header=1;
    repeated contract_info infos = 2;
}

message req_investor_info
{
    int32 requestid=1;
    repeated string investor = 2;
}

message investor_info
{
    string          investorid=1;
    string          brokerid=2;
    string          investortype=3;
    string          investorname=4;
    string          identifiedcardtype=5;
    string          identifiedcardno=6;
    string          country=7;
    string          province=8;
    string          city=9;
    string          birthday=10;
    string          sex=11;
    string          profession=12;
    string          national=13;
    double          capital=14;
    string          capitalcurrency=15;
    string          telephone=16;
    string          address=17;
    string          zipcode=18;
    string          email=19;
    string          memo=20;
    string          openinvestorname=21;
    string          orderinvestorname=22;
    string          fundinvestorname=23;
    string          settleinvestorname=24;
    int32           isactive=25;
    string          departmentid=26;
    string          investorfullname=27;
    int32           isemail=28;
    int32           issms=29;
    int32           isusingotp=30;
    string          clientregion=31;
}

message rsp_investor_info
{
    rsp_head_info header=1;
    repeated investor_info infos = 2;
}

